lib/modules/showImages.js

```diff
@@ -1761,6 +1576,132 @@ addModule('showImages', (module, moduleID) => {
+	async function videoAdvanced(options) {
+		const {
+			fallback,
+			frameRate,
+			playbackRate
+		} = options;
+
+		let {
+			time
+		} = options;
+
+		const template = await RESTemplates.load('video-advanced');
+		const element = template.html(options)[0];
+
+		const vid = element.querySelector('video');
+		const sourceElements = vid.querySelectorAll('source');
+		const ctrlContainer = element.querySelector('.video-advanced-controls');
+
+		const ctrlReverse = ctrlContainer.querySelector('.video-advanced-reverse');
+		const ctrlTogglePause = ctrlContainer.querySelector('.video-advanced-toggle-pause');
+		const ctrlSpeedDecrease = ctrlContainer.querySelector('.video-advanced-speed-decrease');
+		const ctrlSpeedIncrease = ctrlContainer.querySelector('.video-advanced-speed-increase');
+		const ctrlTimeDecrease = ctrlContainer.querySelector('.video-advanced-time-decrease');
+		const ctrlTimeIncrease = ctrlContainer.querySelector('.video-advanced-time-increase');
+
+		const indicatorPosition = element.querySelector('.video-advanced-position');
+
+		const optionsSpeed = element.querySelector('.video-advanced-speed-options');
+
+		const msgError = ctrlContainer.querySelector('.video-advanced-error');
+		const msgSpeed = ctrlContainer.querySelector('.video-advanced-speed');
+		const msgTime = ctrlContainer.querySelector('.video-advanced-time');
+
+		function updatePlayControls() {
+			ctrlTogglePause.innerHTML = vid.paused ? '&#xf16b;' : '&#xf16c;';
+
+			ctrlTimeDecrease.hidden = !vid.paused;
+			ctrlTimeIncrease.hidden = !vid.paused;
+			msgTime.hidden = !vid.paused;
+
+			ctrlSpeedDecrease.hidden = vid.paused;
+			ctrlSpeedIncrease.hidden = vid.paused;
+			msgSpeed.hidden = vid.paused;
+		}
+
+		function reverse() {
+			time = vid.duration - vid.currentTime;
+
+			Array.from(vid.querySelectorAll('source')).forEach(v => {
+				const currentSrc = v.src;
+				v.src = v.dataset.reverse;
+				v.dataset.reverse = currentSrc;
+			});
+
+			vid.load();
+			vid.play();
+
+			ctrlReverse.innerHTML = ctrlReverse.innerText === String.fromCharCode(0xf16d) ? '&#xf169;' : '&#xf16d;';
+		}
+
+		function sourceErrorFallback() {
+			if (fallback) {
+				const fallbackImg = document.createElement('img');
+
+				fallbackImg.src = fallback;
+				fallbackImg.className = 'RESImage';
+
+				element.parentNode.replaceChild(fallbackImg, element);
+				modules['showImages'].makeMediaZoomable(fallbackImg);
+				modules['showImages'].makeMediaMovable(fallbackImg);
+			} else {
+				msgError.hidden = false;
+			}
+		}
+
+		function toggleSpeedOptions() {
+			optionsSpeed.hidden = !optionsSpeed.hidden;
+
+			if (!optionsSpeed.initialized) {
+				optionsSpeed.initialized = true;
+
+				optionsSpeed.addEventListener('click', e => {
+					switch (e.srcElement.dataset['option']) {
+						case 'speed':
+							vid.playbackRate = parseFloat(e.srcElement.dataset['speed'], 10) || 1;
+							break;
+						case 'save':
+							module.options.defaultPlaybackRate.value = vid.playbackRate;
```

Comment:
I agree that the speed selectors are mostly redundant. The idea was to have a way to easily reset the speed, but on second thought might that not be all that necessary.

I am however still interested in finding a way to have the playback rate remembered, either temporarily (till next page load) or persistent (by saving it somewhere). I think the dropdown can be avoided by having an option in the console for whether it should be remembered automatically, and if we find no way to do that, have the user manually set the playback rate manually.